java-tools/OsmAndServer/src/main/java/net/osmand/server/controllers/user/MapApiController.java

```diff
@@ -833,4 +828,112 @@ public String getRegionsByLatlon(@RequestParam("lat") double lat, @RequestParam(
 		regions = osmandRegions.getRegionsToDownload(lat, lon, regions);
 		return gson.toJson(Map.of("regions", regions));
 	}
+
+	@GetMapping(path = {"/generate-shared-url"}, produces = "application/json")
+	public ResponseEntity<String> generateGpxSharedUrl(@RequestParam String name,
+	                                                    @RequestParam String type) {
+		PremiumUserDevicesRepository.PremiumUserDevice dev = checkUser();
+		if (dev == null) {
+			return tokenNotValid();
+		}
+		PremiumUserFilesRepository.UserFile userFile = userdataService.getUserFile(name, type, null, dev);
+		if (userFile == null) {
+			return ResponseEntity.badRequest().body(FILE_NOT_FOUND);
+		}
+		String sharedUrl = userdataService.generateSharedUrl(userFile);
+		return ResponseEntity.ok(gson.toJson(Map.of("sharedUrl", sharedUrl)));
+	}
+
+	@GetMapping(path = {"/gpx/share/{token}"}, produces = "application/json")
```

Comment:
/share/get/{token}

Reply:
What's token?

Reply:
This is a generated unique code to create the shared url
b0244cbc06abda8c69b83722315a for url https://osmand.net/map/gpx/share/b0244cbc06abda8c69b83722315a
It is stored in user_files